### DISCUSSION TITLE
Use of abp.io in a Commercial Web Application - Consequences?

### DISCUSSION THREAD

**Discussion by KieranOnGithub** [abpframework/abp#14512] [General] [answered]
Is it permitted to use the abp.io framework in a commercial project (Saas application on behalf of a customer) without making the project itself freely available in any way?

1. On the one hand, the finished software will be used by the customer in the Azure Cloud and made available to their own customers and employees. On the other hand, local installations of the finished software will be installed on employees' computers. Would this lead to any publication obligation of the created source through the GPL v3? 

2. No changes of any kind are made to the code of the framework itself within the scope of the developments. Any necessary extensions (for example, a derived and extended class) would take place in separate libraries. Would this lead to any publication obligation of the created additional libraries through the GPL v3? 

Thanks in advance for your help and clarification.

**Comment by hikalkan** [comment, marked as answer]
> Is it permitted to use the abp.io framework in a commercial project (Saas application on behalf of a customer) without making the project itself freely available in any way?

Sure. You can use ABP in your commercial, closed-source projects.

> On the one hand, the finished software will be used by the customer in the Azure Cloud and made available to their own customers and employees. On the other hand, local installations of the finished software will be installed on employees' computers. Would this lead to any publication obligation of the created source through the GPL v3?

No problem at all. You can use your software however you want.

> No changes of any kind are made to the code of the framework itself within the scope of the developments. Any necessary extensions (for example, a derived and extended class) would take place in separate libraries. Would this lead to any publication obligation of the created additional libraries through the GPL v3?

If you don't change ABP's source code, and just use it as NuGet packages, you can create your additional libraries. You don't need to publish their source code. You can even sell them. No problem.

**Reply by KieranOnGithub** [reply]
Many Thanks.
